.github/workflows/mlirCI.yml

```diff
@@ -0,0 +1,170 @@
+name: MLIR Workflow
+on:
+  push:
+    paths:
+      - "mlir/**"
+  pull_request:
+    paths:
+      - "mlir/**"
```

Comment:
probably worth adding a workflow trigger here. so that we can manually start the workflow if necesary